R/animint.R

```diff
@@ -649,12 +694,26 @@ saveLayer <- function(l, d, meta){
 saveChunks <- function(x, vars, meta){
   if(is.data.frame(x)){
     if(length(vars) == 0){
+      if(meta$geom %in% c("line", "path", "polygon", "ribbon")) {
+        if("group" %in% names(x)) {
+          # common info for groups, save only once
+          csv.name <- sprintf("%s_chunk_shape.tsv", meta$classed)
+          chunk.shape <- file.path(meta$out.dir, csv.name)
+          if(!file.exists(chunk.shape)){
+            x1 <- x[c("x", "y", "group")]
+            write.table(x1, chunk.shape, quote = FALSE, row.names = FALSE, 
+                        sep = "\t")
+          }
+          # info for each group
+          x2 <- x[, !names(x) %in% c("x", "y")]
+          x <- x2[!duplicated(x2), ]
+          }
```

Comment:
Actually in the WorldBank data viz, the `color` and `clickSelects` columns are different for chunks. For example, the `colour` are different at `78  79  81  84  85  86  87  88  89  90  91  93  94  95  96  97  98  99 101` rows for the first two chunks. @tdhock could you come up with another example for the case you mentioned?